viamonvif/device/device_test.go

```diff
@@ -91,3 +92,106 @@ func readBody(r *http.Request) (string, error) {
 	}
 	return buf.String(), nil
 }
+
+func TestTLSVerificationConfig(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Probably more effort than it is worth, but wondering if we could create a test http server here that uses self-signed TLS cert.

It does seem sufficient to just test that we are skipping verification when dealing with a "private" address.

Reply:
Good call. Not too difficult to add. Added